module-system/sov-state/src/storage.rs

```diff
@@ -140,3 +182,13 @@ impl From<&'static str> for StorageValue {
         }
     }
 }
+
+pub trait NativeStorage: Storage {
+    /// Returns the value corresponding to the key or None if key is absent and a proof to
+    /// get the value. Panics if [`get_with_proof_opt`] returns `None` in place of the proof.
+    fn get_with_proof(
+        &self,
+        key: StorageKey,
+        witness: &Self::Witness,
+    ) -> (Option<StorageValue>, Self::Proof);
```

Comment:
Consider using a wrapper type here. Tuples are harder to document, and a type will incur zero overhead with extended documentation & description

Reply:
+1